spec/lib/miq_automation_engine/service_methods/miq_ae_service_generic_object_spec.rb

```diff
@@ -0,0 +1,36 @@
+module MiqAeServiceGenericObjectSpec
+  describe MiqAeMethodService::MiqAeServiceGenericObject do
+    let(:generic_object)  { FactoryGirl.create(:generic_object) }
+    let(:service_go)      { MiqAeMethodService::MiqAeServiceGenericObject.find(generic_object.id) }
+    let(:service)         { FactoryGirl.create(:service) }
+    let(:service_service) { MiqAeMethodService::MiqAeServiceService.find(service.id) }
+
+    context "#add_to_service" do
+      it "adds a generic object to service_resources of a valid service" do
+        service_go.add_to_service(service_service)
+        expect(service.service_resources[0].resource_id).to eq(generic_object.id)
+        expect(service.service_resources[0].resource_type).to eq(generic_object.class.name)
+      end
+
+      it "raises an error when adding a generic object to an invalid service" do
+        expect { service_go.add_to_service('wrong type') }
+          .to raise_error(ArgumentError, /service must be a MiqAeServiceService/)
+      end
+    end
+
+    context "#remove_from_service" do
+      it "adds a generic object to service_resources of a valid service" do
+        service_go.add_to_service(service_service)
+        expect(service.service_resources.count).to eq(1)
+
+        service_go.remove_from_service(service_service)
+        expect(service.service_resources.count).to eq(0)
+      end
```

Comment:
Include a test that adds two services and remove one to ensure the proper one is removed.